### PR TITLE
Added support to set share tier

### PR DESCRIPTION
**Pull request by gapra-msft**

### CODE REVIEW

File: sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/options/ShareCreateOptions.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import com.azure.storage.file.share.models.ShareAccessTier;
+import com.azure.storage.file.share.models.ShareRequestConditions;
+
+import java.util.Map;
+
+/**
+ * Extended options that may be passed when creating a share.
+ */
+@Fluent
+public class ShareCreateOptions {
```
Comment (kasobol-msft) [inline sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/options/ShareCreateOptions.java:16]: how about CreateShareOptions?

File: sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/options/ShareSetAccessTierOptions.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+import com.azure.core.annotation.Fluent;
+import com.azure.storage.file.share.models.ShareAccessTier;
+import com.azure.storage.file.share.models.ShareRequestConditions;
+
+/**
+ * Extended options that may be passed when setting quota on a share.
+ */
+@Fluent
+public class ShareSetAccessTierOptions {
```
Comment (kasobol-msft) [inline sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/options/ShareSetAccessTierOptions.java:14]: I'd call it ShareAccessTierOptions.

**Review by kasobol-msft — APPROVED** [review]

File: sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/options/ShareCreateOptions.java (hunk 1 above)
Comment (gapra-msft) [reply sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/options/ShareCreateOptions.java:16]: our naming convention has so far been [Resource][Api]Options
Looks like there is a guideline 
https://azure.github.io/azure-sdk/general_design.html#client-interface

This applies to the other comment as well

**Review by rickle-msft — APPROVED** [review]

File: sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/options/ShareCreateOptions.java (hunk 1 above)
Comment (kasobol-msft) [reply sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/options/ShareCreateOptions.java:16]: okay, however being that strict is gonna lead to some interesting names...

**Pull request merged**